File: 프로그래머스/lv1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm> 

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i=0; i<commands.size(); i++) {
        int start = commands[i][0];
        int end = commands[i][1];
        int index = commands[i][2];
        
        vector<int> slice(array.begin() + start - 1, array.begin() + end);
        
        sort(slice.begin(), slice.end());
        
        answer.push_back(slice[index-1]);
        
        // for (int &j: slice) {
        //     std::cout << j << ' ';
        // }
        // cout << endl;
    }
    
    return answer;
}

File: 프로그래머스/lv1/42748. K번째수/K번째수.java
import java.util.*;
/*
[문제 설명]
- i~j까지 정렬 후, k번째 수 구하기
[문제 접근]
- 정렬
*/
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = {};
        List<Integer> list = new ArrayList<>();
        for(int i=0; i<commands.length; i++) {
            // start, end, index는 모두 1부터 시작
            int start = commands[i][0];
            int end = commands[i][1];
            int index = commands[i][2];
            
            int[] slice = Arrays.copyOfRange(array, start-1, end);
            Arrays.sort(slice);
            list.add(slice[index-1]);
        } 
        answer = list.stream().mapToInt(i->i).toArray();
        return answer;
    }
}